File: Example/UNXNavigatorDemo/Feature/Basic/Controllers/ViewController11_ScrollViewWithFullScreen.m
//
//  ViewController11_ScrollViewWithFullScreen.m
//  UNXNavigatorDemo
//
//  Created by Leo Lee on 2020/10/26.
//

#import <UNXNavigator/UNXNavigator.h>

#import "ViewController11_ScrollViewWithFullScreen.h"

@interface ViewController11_ScrollViewWithFullScreen ()

@end

@implementation ViewController11_ScrollViewWithFullScreen

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationItem.title = nil;
    
#if TARGET_OS_MACCATALYST
    CGFloat maxHeight = CGRectGetHeight(self.unx_navigationBar.frame);
    self.tableView.contentInset = UIEdgeInsetsMake(-maxHeight, 0, 0, 0);
#endif
}

- (void)viewWillLayoutSubviews {
    [super viewWillLayoutSubviews];
    
#if !TARGET_OS_MACCATALYST
    CGFloat maxHeight = CGRectGetHeight(self.unx_navigationBar.frame);
    self.tableView.contentInset = UIEdgeInsetsMake(-maxHeight, 0, 0, 0);
#endif
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)unx_barTintColor {
    return [UIColor whiteColor];
}

- (UIColor *)unx_navigationBarBackgroundColor {
    return [UIColor clearColor];
}

- (UIColor *)unx_shadowImageTintColor {
    return [UIColor clearColor];
}

@end

File: Example/UNXNavigatorSwiftDemo/Basic/ViewController11_ScrollViewWithFullScreen.swift
//
//  ViewController11_ScrollViewWithFullScreen.swift
//  UNXNavigatorSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController11_ScrollViewWithFullScreen: ViewController10_ScrollView {

    override func viewDidLoad() {
        super.viewDidLoad()

        navigationItem.title = nil
        
        #if targetEnvironment(macCatalyst)
        let maxHeight = unx_navigationBar.frame.height
        tableView.contentInset = UIEdgeInsets(top: -maxHeight, left: 0.0, bottom: 0.0, right: 0.0)
        #endif
    }
    
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        #if !targetEnvironment(macCatalyst)
        let maxHeight = unx_navigationBar.frame.height
        tableView.contentInset = UIEdgeInsets(top: -maxHeight, left: 0.0, bottom: 0.0, right: 0.0)
        #endif
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var unx_barTintColor: UIColor? {
        return .white
    }
    
    override var unx_navigationBarBackgroundColor: UIColor? {
        return .clear
    }
    
    override var unx_shadowImageTintColor: UIColor? {
        return .clear
    }

}
